Build failure: mdbook-plantuml

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

1. `nix build -f "https://github.com/NixOS/nixpkgs/archive/refs/heads/nixos-unstable.tar.gz" mdbook-plantuml`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/300619155

### Relevant log output

```console
error[E0119]: conflicting implementations of trait `Trait` for type `(dyn Send + Sync + 'static)`
  --> /build/mdbook-plantuml-0.8.0-vendor/traitobject-0.1.0/src/impls.rs:72:1
   |
71 | unsafe impl Trait for ::std::marker::Send + Sync { }
   | ------------------------------------------------ first implementation here
72 | unsafe impl Trait for ::std::marker::Send + Send + Sync { }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `(dyn Send + Sync + 'static)`

error[E0119]: conflicting implementations of trait `Trait` for type `(dyn Send + Sync + 'static)`
  --> /build/mdbook-plantuml-0.8.0-vendor/traitobject-0.1.0/src/impls.rs:73:1
   |
71 | unsafe impl Trait for ::std::marker::Send + Sync { }
   | ------------------------------------------------ first implementation here
72 | unsafe impl Trait for ::std::marker::Send + Send + Sync { }
73 | unsafe impl Trait for ::std::marker::Sync + Send { }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `(dyn Send + Sync + 'static)`

error[E0119]: conflicting implementations of trait `Trait` for type `(dyn Send + Sync + 'static)`
  --> /build/mdbook-plantuml-0.8.0-vendor/traitobject-0.1.0/src/impls.rs:75:1
   |
71 | unsafe impl Trait for ::std::marker::Send + Sync { }
   | ------------------------------------------------ first implementation here
...
75 | unsafe impl Trait for ::std::marker::Sync + Send + Sync { }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `(dyn Send + Sync + 'static)`

For more information about this error, try `rustc --explain E0119`.
error: could not compile `traitobject` (lib) due to 3 previous errors
```

### Additional context

I'm currently `git bisect`ing NixPkgs to find out which change introduced this breakage. This is still in progress. Here's how much I could narrow it down so far: (output of `git bisect log`)
```
git bisect start 'good'
# Status: warte auf guten und schlechten Commit
# good: [dc791e3dbb81095e66e72a30a7f6a10d4d2b88f4] Merge branch 'staging' into staging-next
git bisect good dc791e3dbb81095e66e72a30a7f6a10d4d2b88f4
# Status: warte auf schlechten Commit, 1 guter Commit bekannt
# bad: [9dda4f29b44431c48608c378fcfc5aa303f5e53f] libplacebo: drop the revert needed for mvp (#414326)
git bisect bad 9dda4f29b44431c48608c378fcfc5aa303f5e53f
# good: [3fde79ab4224c0e59adf4dc68d3fbfa420dda637] quaternion-qt5: remove (#410503)
git bisect good 3fde79ab4224c0e59adf4dc68d3fbfa420dda637
# bad: [14604e00fdcb3d00c62560a73c3afa405cb86df6] llvm_{12,13,14,15}: fix build with Python 3.13 (#412548)
git bisect bad 14604e00fdcb3d00c62560a73c3afa405cb86df6
# good: [6b312e2cca63e8577795dacfb7d474dba8c4c70c] Add multiple GPIO related packages by joan2937 (#352308)
git bisect good 6b312e2cca63e8577795dacfb7d474dba8c4c70c
# good: [0d5892c2270c9d3f6053798df5c126101dd3bbcc] gencfsm: fix build error (#411789)
git bisect good 0d5892c2270c9d3f6053798df5c126101dd3bbcc
# bad: [546d1ef5b7cb074a4b013afc076e4c76a6594f32] Merge remote-tracking branch 'origin/staging-next' into staging
git bisect bad 546d1ef5b7cb074a4b013afc076e4c76a6594f32
```
Due to rebuilds of LLVM etc., this may take a while.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.32, NixOS, 25.11 (Xantusia), 25.11.20250607.3e3afe5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.0
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/das-g/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/das-g/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/das-g/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/das-g/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/das-g/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf:/nix/store/fycsc1q7gz9v0yh59l2dcb0l516qlzfp-gnome-settings-daemon-48.1/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/1gy9ax810zrn719m3r71564ckq39yc31-lix-2.93.0/share`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/syvnmj3hhckkbncm94kfkbl76qsdqqj3-source`

### Notify maintainers

@jcouyang @matthiasbeyer

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/md/mdbook-plantuml/package.nix
+++ b/pkgs/by-name/md/mdbook-plantuml/package.nix
@@ -1,33 +1,23 @@
 {
   lib,
   fetchFromGitHub,
-  fetchpatch,
   rustPlatform,
   pkg-config,
   openssl,
 }:
 
-rustPlatform.buildRustPackage rec {
+rustPlatform.buildRustPackage {
   pname = "mdbook-plantuml";
-  version = "0.8.0";
+  version = "0.8.0-unstable-2022-12-28";
 
   src = fetchFromGitHub {
     owner = "sytsereitsma";
     repo = "mdbook-plantuml";
-    tag = "v${version}";
-    hash = "sha256-26epwn6j/ZeMAphiFsrLjS0KIewvElr7V3p/EDr4Uqk=";
+    rev = "c156b53aad6d7bce8479e5406a4a3465c12714ef";
+    hash = "sha256-5/6NQO++MsV7GS69jGkdpkiRhadtQyYZeHreft4h6hQ=";
   };
 
-  cargoPatches = [
-    # https://github.com/sytsereitsma/mdbook-plantuml/pull/60
-    (fetchpatch {
-      name = "update-mdbook-for-rust-1.64.patch";
-      url = "https://github.com/sytsereitsma/mdbook-plantuml/commit/a1c7fdaff65fbbcc086006f6d180b27e180739e7.patch";
-      hash = "sha256-KXFQxogR6SaoX8snsSYMA8gn1FrQVKMl5l8khxB09WE=";
-    })
-  ];
-
-  cargoHash = "sha256-JI5UdLb2ZcPcAIvprVdpQRGMzc0Qu0awIUpMjNyaAPQ=";
+  cargoHash = "sha256-LzzAaWLDODbqGNVeEULLOgrpyLGKzaknIbLjKyF2zBw=";
 
   nativeBuildInputs = [ pkg-config ];
 

